GameDataParser/Parsers/SkillParser.cs

```diff
@@ -13,101 +14,127 @@ namespace GameDataParser.Parsers
-                }
+                    string skillId = Path.GetFileNameWithoutExtension(xmlFile.Name);
+                    SkillMetadata metadata = new SkillMetadata();
+                    List<SkillLevel> skillLevel = new List<SkillLevel>();
 
-                XmlNodeList levels = document.SelectNodes("/ms2/level");
-                foreach (XmlNode node in levels)
-                {
-                    int level = node.Attributes["value"].Value != null ? int.Parse(node.Attributes["value"].Value) : 1;
-                    XmlNode consume = node.SelectSingleNode("consume/stat");
-
-                    int spirit = 0;
-                    int upgradeLevel = 0;
-                    int[] upgradeSkillId = new int[0];
-                    int[] upgradeSkillLevel = new int[0];
+                    metadata.SkillId = int.Parse(skillId);
+                    Debug.Assert(metadata.SkillId > 0, $"Invalid Id {metadata.SkillId} from {skillId}");
 
-                    if (consume.Attributes.GetNamedItem("sp") != null)
-                    {
-                        spirit = int.Parse(consume.Attributes["sp"].Value);
-                    }
+                    XmlDocument document = m2dFile.GetDocument(xmlFile.FileHeader);
 
-                    XmlNode upgrade = node.SelectSingleNode("upgrade");
-                    if (upgrade.Attributes != null)
+                    XmlNodeList basic = document.SelectNodes("/ms2/basic");
+                    foreach (XmlNode node in basic)
                     {
-                        if (upgrade.Attributes.GetNamedItem("level") != null)
-                        {
-                            upgradeLevel = int.Parse(upgrade.Attributes["level"].Value);
-                        }
-                        if (upgrade.Attributes.GetNamedItem("skillIDs") != null)
+                        if (node.Attributes.GetNamedItem("feature") != null)
                         {
-                            upgradeSkillId = Array.ConvertAll(upgrade.Attributes.GetNamedItem("skillIDs").Value.Split(","), int.Parse);
-                        }
-                        if (upgrade.Attributes.GetNamedItem("skillLevels") != null)
-                        {
-                            upgradeSkillLevel = Array.ConvertAll(upgrade.Attributes.GetNamedItem("skillLevels").Value.Split(","), int.Parse);
+                            // Get Rank
+                            metadata.Feature = node.Attributes["feature"].Value;
                         }
                     }
 
-                    XmlNode motion = node.SelectSingleNode("motion/motionProperty");
-
-                    string sequenceName = "";
-                    string motionEffect = "";
-                    string strTagEffects = "";
-
-                    if (motion.Attributes != null)
+                    XmlNodeList levels = document.SelectNodes("/ms2/level");
+                    foreach (XmlNode node in levels)
```

Comment:
Please follow good practice and when iterating over lists use the singular name of the list so `XmlNode level in levels`